include/core/CBoostJsonPoolAllocator.h

```diff
@@ -33,52 +34,52 @@ namespace core {
 //!
 //! Clear the allocator on destruction
 //!
-class CRapidJsonPoolAllocator {
+class CBoostJsonPoolAllocator {
 public:
-    using TDocumentWeakPtr = std::weak_ptr<rapidjson::Document>;
-    using TDocumentPtr = std::shared_ptr<rapidjson::Document>;
+    using TDocumentWeakPtr = std::weak_ptr<boost::json::object>;
+    using TDocumentPtr = std::shared_ptr<boost::json::object>;
     using TDocumentPtrVec = std::vector<TDocumentPtr>;
 
 public:
-    CRapidJsonPoolAllocator()
-        : m_JsonPoolAllocator(m_FixedBuffer, FIXED_BUFFER_SIZE) {}
+    CBoostJsonPoolAllocator() : m_JsonPoolAllocator(m_FixedBuffer) {}
 
-    ~CRapidJsonPoolAllocator() { this->clear(); }
+    ~CBoostJsonPoolAllocator() { this->clear(); }
 
-    void clear() { m_JsonPoolAllocator.Clear(); }
+    void clear() { m_JsonPoolAllocator.release(); }
 
     //! \return document pointer suitable for storing in a container
     //! Note: The API is designed to emphasise that the client does not own the document memory
     //! i.e. The document will be invalidated on destruction of this allocator
     TDocumentWeakPtr makeStorableDoc() {
-        TDocumentPtr newDoc = std::make_shared<rapidjson::Document>(&m_JsonPoolAllocator);
-        newDoc->SetObject();
+        // TODO sort out the allocator pool?
+        //        TDocumentPtr newDoc = std::make_shared<boost::json::object>(&m_JsonPoolAllocator);
```

Comment:
Please can you open an issue listing the nice-to-have followups to this PR. This is one. Changing the method naming to match our style guide is another. There may be some others from previous reviews.

Reply:
Will do. Thanks Dave.